app/grandchallenge/components/models.py

```diff
@@ -713,7 +725,10 @@ def _clean_overlay_segments(self):
             )
 
     def _clean_relative_path(self):
-        if self.is_json_kind:
+        if (
+            self.is_file_kind
+            and self.kind in InterfaceKind.interface_type_json()
+        ):
```

Comment:
I was confused by this at first. Maybe it would be clearer if you added a property `is_json_file_kind`?